### ISSUE TITLE
Typhoon causes crash while transferring autolayout

### ISSUE THREAD

**Issue by Brain89** [appsquickly/Typhoon#513]
Hello,

We have faced with an issue crashing our app

> Fatal Exception: NSUnknownKeyException
> [<NSLayoutConstraint 0x174283cf0> setValue:forUndefinedKey:]: this class is not key value coding-compliant for the key firstItem.

It appears only on devices running iOS 10 (yep, new SDK - new problems). I have attached crash log from Crashlytics. 
As I understand iOS 10 SDK does not support setter for **firstItem** property of NSLayoutConstraint. 

[CrashLog.txt](https://github.com/appsquickly/Typhoon/files/338137/CrashLog.txt)

**Comment by CurlyHeir** [comment]
how to reproduce it?

**Comment by Brain89** [comment]
To reproduce this bug you should add **TyphoonLoadedView** into your project and run it inside iOS X environment.

**Comment by vasilenkoigor** [comment]
Hey @Brain89
Looks strange, because NSLayoutConstraint doesn't changed in iOS 10 SDK, with the exception of added firstAnchor and secondAnchor
https://developer.apple.com/library/prerelease/content/releasenotes/General/iOS10APIDiffs/Objective-C/UIKit.html

However, I sure that crash happens in this place TyphoonViewHelpers
```
[constraint setValue:firstItem forKey:NSStringFromSelector(@selector(firstItem))];
```

@CurlyHeir, If you not found solve problem yet, I can do this tomorrow evening :)

**Comment by Brain89** [comment]
I do not agree that this is a SDK bug. **firstItem** is readonly property (see [docs](https://developer.apple.com/library/ios/documentation/AppKit/Reference/NSLayoutConstraint_Class/index.html#//apple_ref/occ/instp/NSLayoutConstraint/firstItem))

`@property(readonly, assign) id firstItem`

So the code to which you refer uses private API. Apple engineers could change it without any notice.

**Comment by vasilenkoigor** [comment]
@Brain89 
hmm...but wait, this property is readonly since iOS 6...
nope? :)

**Comment by vasilenkoigor** [comment]
Hmm, may be Apple engineers closed  back-door that allows change readonly with KVC😁
Ok, in few days I'll make workaround. 

Thanks for reporting👌

**Comment by Brain89** [comment]
We must reopen this issue. Merged PR breaks this issue https://github.com/appsquickly/Typhoon/pull/464
@CognitiveDisson will fix it.

**Comment by alexgarbarev** [comment]
@Brain89 but how do you think this can be fixed without using "setValue:forKey" since it crashes on iOS10 that way?

**Comment by CognitiveDisson** [comment]
@alexgarbarev Subclass of NSLayoutConstraint fixed this issue. If constraint is it subclass then transfer, else create new constraint. What do you think about this solution?

**Comment by Brain89** [comment]
Well, without creating NSLayoutContraint subclass inside library, every Typhoon user with constraint outlets necessity (like in https://github.com/appsquickly/Typhoon/pull/464) will have to create his own Typhoon fork, where it will be possible to fix this issue (create subclass and improve ViewHelpers in a way @CognitiveDisson wants). Seems very strange too.

**Comment by CognitiveDisson** [comment]
@alexgarbarev If user not use IB to TyphoonLoadedView, then he does not need set subclass. (And all be works like now) But if the user will need this functionality, it will only need to specify the  subclass. 
Another solution is swizzling. (and this solution even more worse)

**Comment by alexgarbarev** [comment]
ok, let's see solution with subclass.

**Comment by etolstoy** [comment]
Fixed in #524
